backend/tests/services/test_dingtalk_wikispace_service.py

```
@@ -0,0 +1,380 @@
+    @pytest.mark.asyncio
+    async def test_pagination_token_extracted(self) -> None:
+        """nextPageToken is extracted from the response envelope."""
+        data = {
+            "items": [{"workspaceId": "WS1", "name": "KB 1"}],
+            "nextPageToken": "tok2",
+        }
+        result, token = DingTalkDocService._parse_list_nodes_result(
+            _make_mcp_result(data)
+        )
+        assert len(result) == 1
+        assert token == "tok2"
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "token"

(S105)